include/vinecopulib/vinecop/implementation/class.ipp

```diff
@@ -630,20 +645,80 @@ Vinecop::get_threshold() const
   return threshold_;
 }
 
+//! @brief sets variable types.
+//! @param var_types a vector of size two specifying the types of the variables,
+//!   e.g., `{"c", "d"}` means first varible continuous, second discrete.
+inline void
+Vinecop::set_var_types(const std::vector<std::string>& var_types)
+{
+  std::stringstream msg;
+  if (var_types.size() > d_) {
+    msg << "more var_types (" << var_types.size() << ")"
+        << "than variables (" << d_ << ")-" << std::endl;
+  }
+  for (auto t : var_types) {
+    if (!tools_stl::is_member(t, { "c", "d" })) {
+      msg << "variable type must be 'c' or 'd'." << std::endl;
+    }
+  }
+  if (!msg.str().empty()) {
+    throw std::runtime_error(msg.str());
+  }
+  var_types_ = var_types;
+  if (pair_copulas_.size() == 0) {
+    return;
+  }
+
+  // set new var_types for all pair-copulas
+  std::vector<std::string> no_types(d_), pair_types(2);
```

Comment:
Those variable names with `no` are always annoying ^^ I'm pretty sure that someone that don't know about natural order would have a hard time here. But I don't really care.

Reply:
I'll change to `natural_types` and `natural_array`